cds-snc/covid-shield-mobile
# Fix A11y focus issues with navigation

### WIP


Fixes various issues surrounding the navigation and auto focus: #689 #704  

**Code changes:**
- Moved `focusOnElement` to the `AccessibilityServiceProvider` from `useAccessibilityAutoFocus`
- Created `useAccessibilityNavigationFocus` for when you need to auto focus an element on screen focus
- Added ability to use auto focus on `ButtonSingleLine` and `Ripple` components
- Refactored all screens that reference the above hooks/functions



**Test cases:**
- Fresh start app with language already selected -> it should read out 'Step 1 of 6' on the onboarding screen ✅ 
- Go to Step 3 and select 'Learn more about how it works' then close the window after it opens -> it should read out the button again ✅ 
- Go to Step 5 and select 'Learn more about privacy' then close window after it opens -> it should read out the button again ✅ 
- Open bottom sheet -> it should read 'Diagnosed with COVID-19?' ✅ 
- Go to code enter screen and select 'Need a one-time key?' then close window -> it should read 'Let people know they may have been exposed' ✅ 
- Open bottom sheet and select 'Change province or territory' then close window -> it should read nothing (this is not implemented yet) ✅ 

**Note: needs to be tested on iOS with VoiceOver**
